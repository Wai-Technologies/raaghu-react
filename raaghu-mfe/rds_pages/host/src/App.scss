//@import "raaghu-react-themes/build/styles/default.css";
// Please dont remove below comment line Adding raaghu-theme in local side.
@import "../../../../raaghu-react-themes/src/styles/default.scss";

.main-body {
    margin-top: 29px;
    position: relative;
}
