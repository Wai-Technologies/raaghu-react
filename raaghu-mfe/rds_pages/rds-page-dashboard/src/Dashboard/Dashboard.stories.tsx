import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import RdsPageDashboard from "./Dashboard";

export default {
    title: "Pages/Dashboard",
    component: RdsPageDashboard,

} as ComponentMeta<typeof RdsPageDashboard>;


const Template: ComponentStory<typeof RdsPageDashboard> = (args) =>
    <RdsPageDashboard {...args} />;


export const Default = Template.bind({});

Default.args = {

};

