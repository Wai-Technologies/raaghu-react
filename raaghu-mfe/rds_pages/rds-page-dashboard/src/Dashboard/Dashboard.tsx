import React, { Suspense } from "react";
import AdminDashboard from "../adminDashboard/adminDashboard";
import TenantDashboard from "../tenantDashboard/tenantDashboard";

export interface DashboardProps { }

const Dashboard = (props: DashboardProps) => {
    const isTenant = localStorage.getItem("abpSession");
    const isImpersonation = localStorage.getItem("isImpersonation") === "true";

    // Check conditions and determine which dashboard to render
    let dashboardToRender;

    if (isTenant) {
        dashboardToRender = <TenantDashboard />;
    } else {
        dashboardToRender = <AdminDashboard user={`${localStorage.getItem("userName")}`} />;
    }

    return (
        <Suspense>
            {dashboardToRender}
        </Suspense>
    );
}


export default Dashboard;
