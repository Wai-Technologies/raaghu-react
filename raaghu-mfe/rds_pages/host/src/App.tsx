import React, { Suspense } from "react";
import "bootstrap/dist/js/bootstrap.bundle.min.js";
import "./App.scss";
import AuthHigherComponent from "./AuthHigherComponent";

const App = () => {
    return (
        <Suspense>
            <AuthHigherComponent></AuthHigherComponent>
        </Suspense>
    );
};

export default App;
