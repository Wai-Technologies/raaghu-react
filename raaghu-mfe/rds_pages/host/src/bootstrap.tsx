import App from "./App";
import ReactDOM from "react-dom/client";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../../libs/public.api";
import { Provider } from "react-redux";

(ReactDOM as any).createRoot(document.getElementById("root")).render(
    <Provider store={store}>
        <div>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </div>
    </Provider>
);

