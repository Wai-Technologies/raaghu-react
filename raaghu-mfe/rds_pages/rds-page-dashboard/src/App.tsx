import { Suspense } from "react";
import React from "react";
import Dashboard from "./Dashboard/Dashboard";

const App = () => (
    <Suspense>
        <Dashboard></Dashboard>
    </Suspense>
);

export default App;
