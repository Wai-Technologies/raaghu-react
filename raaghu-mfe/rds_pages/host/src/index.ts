import("./bootstrap");
import "../../../web.config";
