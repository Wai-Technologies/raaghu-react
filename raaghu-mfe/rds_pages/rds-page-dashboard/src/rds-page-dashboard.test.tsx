import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../../libs/state-management";
import Dashboard from "./Dashboard/Dashboard";
import TenantDashboard from "./tenantDashboard/tenantDashboard";
import AdminDashboard from "./adminDashboard/adminDashboard";
import { RdsBigNumber, RdsBooleanChart, RdsDoughnutChart, RdsLineChart, RdsMap, RdsProgressBar, RdsRadarChart, RdsTable, RdsWidget } from "../../../../raaghu-elements/src";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
        t: (key: string) => key,
    }),
}));

jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
}));

global.location = {
    pathname: "../../../libs/proxy/core",
  } as any;
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));
  jest.mock("../../../../raaghu-components/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
  }));
  jest.mock("../../../../raaghu-elements/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
  }));
  jest.mock("../../../libs/state-management/index.ts", () => ({
    store: {
      getState: jest.fn().mockReturnValue({
        persistedReducer: {
          configureStore: {
            configuration: {
              auth: {
                grantedPolicies: {},
              },
            },
          },
        },
      }),
      dispatch: jest.fn(),
      subscribe: jest.fn(),
    },
  }));
  
  // Mock the rds-elements and rds-components
  jest.mock("../../../../raaghu-elements/src", () => ({
    RdsButton: jest.fn(),
    RdsOffcanvas: jest.fn(),
    RdsAlert: jest.fn(),
    RdsSearch: jest.fn(),
    RdsModal: jest.fn(),
    RdsChartLine: jest.fn(),
    RdsChartDoughnut: jest.fn(),
    RdsChartBoolean: jest.fn(),
    RdsChartBar: jest.fn(),
    RdsWidget: jest.fn(),
    RdsLineChart: jest.fn(),
    RdsMap: jest.fn(),
    RdsBigNumber: jest.fn(),
    RdsDoughnutChart: jest.fn(),
    RdsBooleanChart: jest.fn(),
    RdsTable: jest.fn(),
    RdsProgressBar: jest.fn(),
    RdsRadarChart: jest.fn(),
  }));
  
  jest.mock("../../../../raaghu-components/src", () => ({
    RdsCompApiScopeBasicResource: jest.fn(),
    RdsCompDatatable: jest.fn(),
    RdsCompAlertPopup: jest.fn(),
    RdsOffcanvas:jest.fn()
  }));
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useDispatch: jest.fn(),
  }));
  
  // Mock localStorage in your test setup
  const localStorageMock: Storage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
    length: 0, // Add length property
    key: jest.fn(), // Add key method
  };
  

  global.localStorage = localStorageMock;

  jest.mock("chart.js", () => ({
    ...(jest.requireActual("chart.js") as object),
    ...{
      Line: jest.fn(),
    },
  }));
   
  jest.mock("../../../../raaghu-elements/src/rds-chart-line/rds-chart-line.tsx", () => {
    return jest.fn(() => <div data-testid="mocked-rds-line-chart" />);
  });
   
  jest.mock(
    "../../../../raaghu-elements/src/rds-chart-doughnut/rds-chart-doughnut.tsx",
    () => {
      return jest.fn(() => <div data-testid="mocked-rds-line-chart" />);
    }
  );
   
  jest.mock(
    "../../../../raaghu-elements/src/rds-chart-boolean/rds-chart-boolean.tsx",
    () => {
      return jest.fn(() => <div data-testid="mocked-rds-line-chart" />);
    }
  );
   
  jest.mock(
      "../../../../raaghu-elements/src/rds-chart-bar/rds-chart-bar.tsx",
      () => {
        return jest.fn(() => <div data-testid="mocked-rds-line-chart" />);
      }
    );
   
  jest.mock("bootstrap", () => ({
    Tooltip: jest.fn().mockImplementation(() => ({
      dispose: jest.fn(),
    })),
  }));


  describe("Dashboard", () => {
it("renders Dashboard correctly", async() => {
        // Mock the useDispatch hook
    const mockDispatch = jest.fn();
    (mockDispatch as jest.Mock).mockReturnValue(Promise.resolve({
      payload: {
        totalCount: 10,
      },
    }));

    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    render(
        <Provider store={store}>
            <Dashboard />
        </Provider>
    );
    await Promise.resolve();
});

it("Check Tenant-dashbord page fucntionality", async() => {
 const handleButtonClick =jest.fn();
 const mockDispatch = jest.fn();
 (mockDispatch as jest.Mock).mockReturnValue(Promise.resolve({
   payload: {
     totalCount: 10,
   },
 }));

 (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
const tenantDash = render(
     <Provider store={store}>
         <Dashboard />
         <TenantDashboard />
         <RdsLineChart  labels={["12am", "4am", "8am", "12pm", "4pm", "8pm"]} options={undefined} dataSets={[]} id={"linechart"} />
        <RdsWidget headerTitle={"License"}  handleButtonClick={handleButtonClick}/>
        <RdsBigNumber bigNumber="10"></RdsBigNumber>
        <RdsRadarChart   labels={["Jan", "Feb", "Mar", "Apr", "May", "June", "July"]}options={undefined} dataSets={[]} id={"newRadar"} />
        <RdsWidget headerTitle={"Profit Share"} />
        <RdsBigNumber bigNumber="$39,330"></RdsBigNumber>
        <RdsDoughnutChart  labels={['Total Sales - 85%', 'Revenue - 25%', 'Expenses - 15%']} options={undefined} dataSets={[]} id={"doughnutchart"} />
        <RdsBooleanChart labels={["Total Clients Called" ]} options={undefined} dataSets={[]} id={"Boolean1"} />
        <RdsTable headerDatas={[]} tableDatas={[]} />
         <RdsProgressBar colorVariant={"primary"} progressWidth={40} role={"single"} />

     </Provider>
 );
 expect(tenantDash).toMatchSnapshot();
 await Promise.resolve();
});

it("Check Admin-dashbord page fucntionality", async() => {
  const handleButtonClick = jest.fn();
  const onBackSide = jest.fn();
  const mockDispatch = jest.fn();
  (mockDispatch as jest.Mock).mockReturnValue(Promise.resolve({
    payload: {
      totalCount: 10,
    },
  }));
 
  (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
 const adminDash =  render(
      <Provider store={store}>
          <Dashboard />
          <AdminDashboard user={"abc"} />
          <RdsWidget headerTitle={"Daily Summary"}  handleButtonClick={handleButtonClick}/>
          <RdsLineChart labels={["12am", "4am", "8am", "12pm", "4pm", "8pm"]} options={undefined} dataSets={[]} id={"linechart"} />
          <RdsWidget headerTitle={"Sales"} onIconClick={onBackSide}/>
          <RdsMap mapList={undefined} color={"purple"} />
      </Provider>
  );
  expect(adminDash).toMatchSnapshot();
  await Promise.resolve();
 });

afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
});