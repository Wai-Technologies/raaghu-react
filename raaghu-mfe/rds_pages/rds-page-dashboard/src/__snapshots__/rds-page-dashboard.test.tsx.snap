// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Check Admin-dashbord page fucntionality 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dark dashboard"
      >
        <div
          class="row"
        >
          <div
            class="col-xl-6 col-lg-12 col-md-12 d-cus-none"
          />
          <div
            class="col-xl-3 col-lg-6 col-md-12"
          >
            <div
              class="FlipCard"
            >
              <div
                class="innerCard innerCardFront"
              >
                <div
                  class="frontSide"
                />
                <div
                  class="backSide"
                />
              </div>
            </div>
          </div>
          <div
            class="col-xl-3 col-lg-6 col-md-12"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12 col-lg-12 col-xl-6"
          >
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <div />
                <div />
              </div>
              <div
                class="col-md-6"
              />
            </div>
          </div>
          <div
            class="col-md-12 col-lg-12 col-xl-6"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          />
        </div>
      </div>
      <div
        class="dark dashboard"
      >
        <div
          class="row"
        >
          <div
            class="col-xl-6 col-lg-12 col-md-12 d-cus-none"
          />
          <div
            class="col-xl-3 col-lg-6 col-md-12"
          >
            <div
              class="FlipCard"
            >
              <div
                class="innerCard innerCardFront"
              >
                <div
                  class="frontSide"
                />
                <div
                  class="backSide"
                />
              </div>
            </div>
          </div>
          <div
            class="col-xl-3 col-lg-6 col-md-12"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12 col-lg-12 col-xl-6"
          >
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <div />
                <div />
              </div>
              <div
                class="col-md-6"
              />
            </div>
          </div>
          <div
            class="col-md-12 col-lg-12 col-xl-6"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dark dashboard"
    >
      <div
        class="row"
      >
        <div
          class="col-xl-6 col-lg-12 col-md-12 d-cus-none"
        />
        <div
          class="col-xl-3 col-lg-6 col-md-12"
        >
          <div
            class="FlipCard"
          >
            <div
              class="innerCard innerCardFront"
            >
              <div
                class="frontSide"
              />
              <div
                class="backSide"
              />
            </div>
          </div>
        </div>
        <div
          class="col-xl-3 col-lg-6 col-md-12"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12 col-lg-12 col-xl-6"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div />
              <div />
            </div>
            <div
              class="col-md-6"
            />
          </div>
        </div>
        <div
          class="col-md-12 col-lg-12 col-xl-6"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        />
      </div>
    </div>
    <div
      class="dark dashboard"
    >
      <div
        class="row"
      >
        <div
          class="col-xl-6 col-lg-12 col-md-12 d-cus-none"
        />
        <div
          class="col-xl-3 col-lg-6 col-md-12"
        >
          <div
            class="FlipCard"
          >
            <div
              class="innerCard innerCardFront"
            >
              <div
                class="frontSide"
              />
              <div
                class="backSide"
              />
            </div>
          </div>
        </div>
        <div
          class="col-xl-3 col-lg-6 col-md-12"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12 col-lg-12 col-xl-6"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div />
              <div />
            </div>
            <div
              class="col-md-6"
            />
          </div>
        </div>
        <div
          class="col-md-12 col-lg-12 col-xl-6"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Dashboard Check Tenant-dashbord page fucntionality 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dark dashboard"
      >
        <div
          class="row"
        >
          <div
            class="col-xl-6 col-lg-12 col-md-12 d-cus-none"
          />
          <div
            class="col-xl-3 col-lg-6 col-md-12"
          >
            <div
              class="FlipCard"
            >
              <div
                class="innerCard innerCardFront"
              >
                <div
                  class="frontSide"
                />
                <div
                  class="backSide"
                />
              </div>
            </div>
          </div>
          <div
            class="col-xl-3 col-lg-6 col-md-12"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12 col-lg-12 col-xl-6"
          >
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <div />
                <div />
              </div>
              <div
                class="col-md-6"
              />
            </div>
          </div>
          <div
            class="col-md-12 col-lg-12 col-xl-6"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          />
        </div>
      </div>
      <div
        class="dark dashboard"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-6 col-md-12 d-cus-none"
          />
          <div
            class="col-lg-3 col-md-12"
          />
          <div
            class="col-lg-3 col-md-12"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12 col-lg-6"
          >
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <div />
                <div />
              </div>
              <div
                class="col-md-6"
              />
            </div>
          </div>
          <div
            class="col-md-12 col-lg-6"
          />
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dark dashboard"
    >
      <div
        class="row"
      >
        <div
          class="col-xl-6 col-lg-12 col-md-12 d-cus-none"
        />
        <div
          class="col-xl-3 col-lg-6 col-md-12"
        >
          <div
            class="FlipCard"
          >
            <div
              class="innerCard innerCardFront"
            >
              <div
                class="frontSide"
              />
              <div
                class="backSide"
              />
            </div>
          </div>
        </div>
        <div
          class="col-xl-3 col-lg-6 col-md-12"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12 col-lg-12 col-xl-6"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div />
              <div />
            </div>
            <div
              class="col-md-6"
            />
          </div>
        </div>
        <div
          class="col-md-12 col-lg-12 col-xl-6"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        />
      </div>
    </div>
    <div
      class="dark dashboard"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-6 col-md-12 d-cus-none"
        />
        <div
          class="col-lg-3 col-md-12"
        />
        <div
          class="col-lg-3 col-md-12"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12 col-lg-6"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div />
              <div />
            </div>
            <div
              class="col-md-6"
            />
          </div>
        </div>
        <div
          class="col-md-12 col-lg-6"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
